backend/open_webui/functions.py

```
@@ -0,0 +1,316 @@
+                    "name": pipe.name,
+                    "object": "model",
+                    "created": pipe.created_at,
+                    "owned_by": "openai",
+                    "pipe": pipe_flag,
+                }
+            )
+
+    return pipe_models
+
+
+async def generate_function_chat_completion(
```

Ruff (0.8.2)

`generate_function_chat_completion` is too complex (38 > 10)

(C901)